components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/db/SitePermissionsDao.kt

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions.db
+
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Delete
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy
+import android.arch.persistence.room.Query
+
+/**
+ * Internal dao for accessing and modifying sitePermissions in the database.
+ */
+@Dao
+internal interface SitePermissionsDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertOrUpdateSitePermission(entity: SitePermissionsEntity): Long
```

Comment:
This is convenient but can also be the source of bugs if you start "replacing" values when you actually expect to "insert" (e.g. Insert "Allow microphone" replaces "Allow camera" in database, instead of updating the existing entry).

Reply:
👍